internal/importer/importer.go

```diff
@@ -216,37 +215,75 @@ func (i *azureImporter) importBranchesWritesProblem(defaultBranchRef string, bra
 //
 // This relies on the "cannot-change-group" being removed, as was done in
 // wharf-api v4.2.0: https://github.com/iver-wharf/wharf-api/pull/55
-func (i *azureImporter) createOrUpdateWharfProject(orgName string, repo azureapi.Repository, buildDef string) (wharfapi.Project, error) {
-	var project wharfapi.Project
-	searchResults, err := i.wharf.SearchProject(wharfapi.Project{
-		Name:       repo.Project.Name,
-		GroupName:  orgName,
-		ProviderID: i.provider.ProviderID,
+func (i *azureImporter) createOrUpdateWharfProject(orgName string, repo azureapi.Repository, buildDef string) (response.Project, error) {
+	groupName := fmt.Sprintf("%s/%s", orgName, repo.Project.Name)
+
+	var existingProject response.Project
+	search := wharfapi.ProjectSearch{
+		Name:       &repo.Name,
+		GroupName:  &groupName,
+		ProviderID: &i.provider.ProviderID,
+	}
+	searchResults, err := i.wharf.GetProjectList(search)
+	if err != nil {
+		log.Error().
+			WithError(err).
+			WithString("name", *search.Name).
+			WithString("groupName", *search.GroupName).
+			WithUint("providerId", *search.ProviderID).
+			Message("Unable to search for existing project.")
+		return existingProject, err
+	}
+	if len(searchResults.List) > 0 {
+		existingProject = searchResults.List[0]
+		updatedProject := request.ProjectUpdate{
+			Name:            repo.Name,
+			TokenID:         i.token.TokenID,
+			GroupName:       groupName,
+			BuildDefinition: buildDef,
+			Description:     repo.Project.Description,
+			ProviderID:      i.provider.ProviderID,
+			GitURL:          repo.SSHURL,
+		}
+		return i.wharf.UpdateProject(existingProject.ProjectID, updatedProject)
+	}
+
+	createdProject, err := i.wharf.CreateProject(request.Project{
+		Name:            repo.Name,
+		TokenID:         i.token.TokenID,
+		GroupName:       groupName,
+		BuildDefinition: buildDef,
+		Description:     repo.Project.Description,
+		ProviderID:      i.provider.ProviderID,
+		GitURL:          repo.SSHURL,
+		RemoteProjectID: repo.Project.ID,
 	})
-	if err != nil && len(searchResults) > 0 {
-		project = searchResults[0]
+
+	if err != nil {
+		log.Error().
+			WithError(err).
+			WithString("name", repo.Project.Name).
+			WithString("groupName", groupName).
+			WithString("gitURL", repo.SSHURL).
+			WithUint("providerId", *search.ProviderID).
+			Message("Unable to create project.")
+		return response.Project{}, err
 	}
-	project.Name = repo.Name
-	project.TokenID = i.token.TokenID
-	project.GroupName = fmt.Sprintf("%s/%s", orgName, repo.Project.Name)
-	project.BuildDefinition = buildDef
-	project.Description = repo.Project.Description
-	project.ProviderID = i.provider.ProviderID
-	project.GitURL = repo.SSHURL
-	return i.wharf.PutProject(project)
+
+	return createdProject, nil
 }
```

Comment:
This method became incredibly bloated due to the removal of creation in the `PUT` endpoints.

I would like to clean up stuff like this in a different PR though, as I think it'd be easier with some logic/flow changes and that feels out-of-scope for this PR.

Reply:
I'm personally fine with this.